integration-tests/src/tests/client/process_blocks.rs

```diff
@@ -2670,6 +2670,107 @@ fn test_wasmer2_upgrade() {
+
+    // Prepare TestEnv with a contract at the old protocol version.
+    let mut env = {
+        let epoch_length = 5;
+        let mut genesis =
+            Genesis::test(vec!["test0".parse().unwrap(), "test1".parse().unwrap()], 1);
+        genesis.config.epoch_length = epoch_length;
+        genesis.config.protocol_version = old_protocol_version;
+        let chain_genesis = ChainGenesis::new(&genesis);
+        TestEnv::builder(chain_genesis)
+            .runtime_adapters(vec![Arc::new(
+                nearcore::NightshadeRuntime::test_with_runtime_config_store(
+                    Path::new("../../../.."),
+                    create_test_store(),
+                    &genesis,
+                    TrackedConfig::new_empty(),
+                    RuntimeConfigStore::new(None),
+                ),
+            ) as Arc<dyn RuntimeAdapter>])
+            .build()
+    };
+
+    let signer = InMemorySigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+    let tx = Transaction {
+        signer_id: "test0".parse().unwrap(),
+        receiver_id: "test0".parse().unwrap(),
+        public_key: signer.public_key(),
+        actions: vec![Action::AddKey(AddKeyAction {
+            public_key: signer.public_key(),
+            access_key: AccessKey {
+                nonce: 1,
+                permission: AccessKeyPermission::FunctionCall(FunctionCallPermission {
+                    allowance: None,
+                    receiver_id: "#".to_string(),
+                    method_names: vec![],
+                }),
+            },
+        })],
+        nonce: 0,
+        block_hash: CryptoHash::default(),
+    };
+
+    // Run the transaction & collect the logs.
+    {
+        let tip = env.clients[0].chain.head().unwrap();
+        let signed_transaction =
+            Transaction { nonce: 10, block_hash: tip.last_block_hash, ..tx.clone() }.sign(&signer);
+        let res = env.clients[0].process_tx(signed_transaction, false, false);
+        assert_eq!(res, NetworkClientResponses::ValidTx);
+        for i in 0..3 {
+            env.produce_block(0, tip.height + i + 1);
+        }
+    };
+
+    // Move to the new protocol version.
+    {
+        let tip = env.clients[0].chain.head().unwrap();
+        let epoch_id = env.clients[0]
+            .runtime_adapter
+            .get_epoch_id_from_prev_block(&tip.last_block_hash)
+            .unwrap();
+        let block_producer =
+            env.clients[0].runtime_adapter.get_block_producer(&epoch_id, tip.height).unwrap();
+        let mut block = env.clients[0].produce_block(tip.height + 1).unwrap().unwrap();
+        set_block_protocol_version(&mut block, block_producer, new_protocol_version);
+        let (_, res) = env.clients[0].process_block(block.clone().into(), Provenance::NONE);
+        assert!(res.is_ok());
+        for i in 0..6 {
+            env.produce_block(0, tip.height + i + 2);
+        }
+    }
+
+    // Re-run the transaction.
```

Comment:
and comment that we expect this transaction to fail, as this is the protocol version that enforces the account ids.